python/cudf/cudf/core/column/column.py

```diff
@@ -1486,7 +1454,12 @@ def _has_any_nan(arbitrary: pd.Series | np.ndarray) -> bool:
 def column_empty(
     row_count: int, dtype: Dtype = "object", masked: bool = False
```

Comment:
Is the `masked` parameter no longer used?

I also want to raise a concern: the nullability of a column is conveyed by the presence/absence of a mask. While a column with zero rows has no data, the nullability is moreso a part of its `dtype` which is independent of the row count. We have to track nullability carefully for things like round-tripping Parquet data (where nullability is part of the schema). I would check carefully if this changes any I/O behaviors for zero-row files -- we may not have adequate test coverage today. libcudf/cudf are notoriously weak/imprecise in this area and I want to get stricter, not less strict, over time.

Reply:
> Is the masked parameter no longer used?

Correct.

> I would check carefully if this changes any I/O behaviors for zero-row files -- we may not have adequate test coverage today.

I added a small, zero row roundtrip parquet unit test in [34fbc8a](https://github.com/rapidsai/cudf/pull/16715/commits/34fbc8a6502aa28e7266663276d05401b3d78abd) to validate this was unaffected. Agreed that it would be good to make libcudf/cudf stricter w.r.t. mask presence, and at least this PR makes it more consistent for zero row vs all null columns generated in cudf

Reply:
Can we remove the `masked` parameter now that it is unused?

Reply:
I want to remove this parameter in a follow up since there are some usages of `masked` in other RAPIDS libraries. I'll address those first (after this PR) before opening up a PR to remove it here